docs: add Python SDK parity documentation for observation formatting (#588)
enyst/OpenHands-Tab: packages/agent-sdk-ts/docs/python-parity.md

  return JSON.stringify(result, null, 2);
}
```

**Problems:**
- Formatting logic external to tools, not owned by them
- Generic fallback leaks internal fields (like `summary`) to the LLM
- Adding a new tool requires updating `toolMessageFormatting.ts`

### Migration plan

1. Add `formatForLLM?(result: TResult): string` to `ToolDefinition` interface
2. Implement in each tool (move logic from `toolMessageFormatting.ts`)
3. Update `Agent.ts` to use tool's formatter
4. Remove generic JSON.stringify fallback (or make it throw)

### Source references
- Python: `openhands-sdk/openhands/sdk/tool/schema.py` (Observation.to_llm_content); `openhands-sdk/openhands/sdk/event/llm_convertible/observation.py` (ObservationEvent.to_llm_message); `openhands-tools/openhands/tools/terminal/definition.py` (TerminalObservation)
- TypeScript: `packages/agent-sdk-ts/src/sdk/types/tools.ts` (ToolDefinition); `packages/agent-sdk-ts/src/sdk/runtime/toolMessageFormatting.ts`; `packages/agent-sdk-ts/src/sdk/runtime/Agent.ts`

## Quick checklist for parity work
- **Observation formatting (#587)**: Add `formatForLLM()` to tools so observations own their LLM representation (like Python's `to_llm_content`).

> - Implement workspace factory/base with remote support, path validation, git helpers, and richer command metadata.

Author: ```suggestion
- Implement workspace factory/base with remote support, and maybe richer command metadata.
```

### Migration plan

1. Add `formatForLLM?(result: TResult): string` to `ToolDefinition` interface
2. Implement in each tool (move logic from `toolMessageFormatting.ts`)
3. Update `Agent.ts` to use tool's formatter
4. Remove generic JSON.stringify fallback (or make it throw)

### Source references
- Python: `openhands-sdk/openhands/sdk/tool/schema.py` (Observation.to_llm_content); `openhands-sdk/openhands/sdk/event/llm_convertible/observation.py` (ObservationEvent.to_llm_message); `openhands-tools/openhands/tools/terminal/definition.py` (TerminalObservation)
- TypeScript: `packages/agent-sdk-ts/src/sdk/types/tools.ts` (ToolDefinition); `packages/agent-sdk-ts/src/sdk/runtime/toolMessageFormatting.ts`; `packages/agent-sdk-ts/src/sdk/runtime/Agent.ts`

## Quick checklist for parity work
- **Observation formatting (#587)**: Add `formatForLLM()` to tools so observations own their LLM representation (like Python's `to_llm_content`).
- Implement workspace factory/base with remote support, and maybe richer command metadata.
- Extend conversations with visualizer/stuck-detection hooks, callback stacks, and remote workspace helpers.

> - Augment agent with condenser/security analyzers, persisted state replay, and expanded confirmation policies.

Author: ```suggestion
- Augment agent with what is missing from condenser/security analyzers, and test confirmation policies.
```
